app/imports/components/posts/NewPost.jsx

```diff
@@ -106,25 +115,15 @@ export class NewPost extends Component {
       ctx.drawImage(this.phantomImage, 0, 0);
 
       // export base64
-      this.setState({
-        fixedOrientation: true,
-        photoRotatedSrc: canvas.toDataURL(),
-        finalWidth: canvas.width,
-        finalHeight: canvas.height,
-      });
-    } else {
-      this.setState({
-        fixedOrientation: true,
-        photoRotatedSrc: this.state.originalPhoto,
-        finalWidth: width,
-        finalHeight: height,
-      });
+      let mimeType = Meteor.settings.public.imageCompression.mimeType;
+      this.phantomImage.src = canvas.toDataURL(mimeType, 1);
```

Comment:
most of this functions are un-pure, they don't necessarily return a value and have side-effects. i think this should stay like this in order to keep `resizeImage()` and `fixImageRotation()` consistent